raft/src/main/scala/zio/raft/Raft.scala

```diff
@@ -907,6 +908,72 @@ class Raft[S, A <: Command](
       res <- promiseArg.await
     yield (res)
 
+  private def validateLeadership(leader: Leader[S]): ZIO[Any, NotALeaderError, Unit] =
+    for
+      now <- zio.Clock.instant
+      peersRequiringHeartbeat = peers.filter(peer => leader.heartbeatDue.due(now, peer))
+      
+      _ <- ZIO.logDebug(
+        s"memberId=${this.memberId} validateLeadership: ${peersRequiringHeartbeat.length}/${peers.length} peers requiring heartbeat"
+      )
+      
+      // TODO (eran): TBD with Doron: as per Raft paper, we might want to always send heartbeat and not rely on the heartbeat due time
+      
+      // If majority of peers don't require heartbeat, leadership is likely valid
+      // peersRequiringHeartbeat.length < numberOfServers / 2 means:
+      // - peers not requiring heartbeat + self > numberOfServers / 2 (majority)
+      _ <- if peersRequiringHeartbeat.length < numberOfServers / 2 then
+        ZIO.logDebug(s"memberId=${this.memberId} Leadership validation: recent heartbeats sufficient")
+        *> ZIO.unit
+      else
+        // Need to send heartbeats to validate leadership
+        for
+          _ <- ZIO.logDebug(
+            s"memberId=${this.memberId} Leadership validation: sending heartbeats to ${peersRequiringHeartbeat.mkString(", ")}"
+          )
+          
+          // Reuse existing sendHeartbeatRule for each peer requiring heartbeat
+          _ <- ZIO.foreachDiscard(peersRequiringHeartbeat)(sendHeartbeatRule)
+          
+          // Wait briefly for responses to validate leadership
+          // If we don't receive majority responses within timeout, assume leadership lost
+          _ <- zio.Clock.sleep(rpcTimeout)
```

Comment:
@somdoron also want to discuss this with you.
This is naive implementation, I don't like this (sleep), the alternative adds more complexity: calling the heartbeat rule, adding some promise to wait for responses (a command?) and waiting on the promise with some timeout.

curious on your thoughts